### PR TITLE
Refactor tests under `events` package

### PR DESCRIPTION
**Pull request by russjones**
Refactored all tests under `lib/events` to use testify instead of gocheck.

### CODE REVIEW

File: lib/events/auditlog_test.go [hunk 1]
```diff
@@ -110,53 +69,66 @@ func (a *AuditTestSuite) TestLogRotation(c *check.C) {
...
+		err := alog.EmitAuditEvent(context.TODO(), event)
+		require.NoError(t, err)
 		logfile := alog.localLog.file.Name()
 
 		// make sure that file has the same date as the event
 		dt, err := parseFileTime(filepath.Base(logfile))
-		c.Assert(err, check.IsNil)
-		c.Assert(dt, check.Equals, time.Date(now.Year(), now.Month(), now.Day(), 0, 0, 0, 0, now.Location()))
+		require.NoError(t, err)
+		require.Equal(t, dt, time.Date(now.Year(), now.Month(), now.Day(), 0, 0, 0, 0, now.Location()))
```
Comment (zmb3) [inline lib/events/auditlog_test.go:79]: Technically the order of the expected/actual values in the assertion is wrong, which is fine other than the confusing error message you get if things fail.

File: lib/events/dynamoevents/dynamoevents_test.go [hunk 2]
```diff
@@ -74,71 +69,82 @@ func (s *suiteBase) SetUpSuite(c *check.C) {
...
+		return tt.log.deleteTable(context.Background(), tt.log.Tablename, true)
 	}
+
+	return nil
 }
 
-type DynamoeventsSuite struct {
-	suiteBase
+func TestPagination(t *testing.T) {
+	tt := setupDynamoContext(t)
```
Comment (zmb3) [inline lib/events/dynamoevents/dynamoevents_test.go:93]: The guideline I've been giving is that constructors and other test helpers that take a `*testing.T` should register any necessary cleanups.

So in this case, move the `t.Cleanup()` call into `setupDynamoContext()`, then the callers don't need to worry about it.

File: lib/events/dynamoevents/dynamoevents_test.go [hunk 3]
```diff
@@ -147,27 +153,26 @@ func (s *DynamoeventsSuite) TestSizeBreak(c *check.C) {
 		}
 	}
 
-	lastTime := s.Clock.Now().UTC().Add(time.Hour)
+	lastTime := tt.suite.Clock.Now().UTC().Add(time.Hour)
 
 	for _, event := range events {
-		c.Assert(event.GetTime().Before(lastTime), check.Equals, true)
+		require.Equal(t, event.GetTime().Before(lastTime), true)
```
Comment (zmb3) [inline lib/events/dynamoevents/dynamoevents_test.go:159]: A rewrite rule for `require.Equal(t, a, true) -> require.True(t, a)` would clean a lot of these up.

File: lib/events/firestoreevents/firestoreevents_test.go [hunk 4]
```diff
@@ -61,53 +57,80 @@ func (s *FirestoreeventsSuite) SetUpSuite(c *check.C) {
...
+	require.NoError(t, err)
 }
 
-func (s *FirestoreeventsSuite) TestSessionEventsCRUD(c *check.C) {
-	s.SessionEventsCRUD(c)
+func TestSessionEventsCRUD(t *testing.T) {
+	tt := setupFirestoreContext(t)
+	t.Cleanup(func() {
+		tt.Cleanup(t)
+		tt.Close()
```
Comment (zmb3) [inline lib/events/firestoreevents/firestoreevents_test.go:103]: Move these cleanup registrations into `setupFirestoreContext`

File: lib/events/firestoreevents/firestoreevents_test.go [hunk 5]
```diff
@@ -61,53 +59,69 @@ func (s *FirestoreeventsSuite) SetUpSuite(c *check.C) {
...
+	if tt.log != nil {
+		err := tt.log.Close()
+		require.NoError(t, err)
+	}
 }
 
-func (s *FirestoreeventsSuite) TestSessionEventsCRUD(c *check.C) {
-	s.SessionEventsCRUD(c)
+func TestSessionEventsCRUD(t *testing.T) {
+	tt := setupFirestoreContext(t)
```
Comment (r0mant) [inline lib/events/firestoreevents/firestoreevents_test.go:103]: Previously, firestore context was set up once per suite, now it's setup in each test. Is it intensive operation? I think we've been recently moving towards the opposite - i.e. for app/db tests we now set up all infra once and run all tests against it (at least, for integration tests). Maybe do the same here?

**Review by r0mant — APPROVED** [review]
Left a suggestion but I'm ok either way.

File: lib/events/firestoreevents/firestoreevents_test.go (hunk 5 above)
Comment (russjones) [reply lib/events/firestoreevents/firestoreevents_test.go]: Refactored so it works more like the old tests (broke out setting up of the suite then setting up of a test). Not sure if this reads any better to be honest.

**Review by rosstimothy — APPROVED** [review]

**Review by zmb3 — APPROVED** [review]

**Pull request merged**